packages/pyxis-react/src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,114 @@
+import { cloneElement, FC, ReactElement } from 'react';
+import classNames from 'classnames';
+import Header, { HeaderProps } from './Header';
+import Footer, { FooterProps } from './Footer';
+import Content from './Content';
+
+const getBackdropClasses = (isOpen: boolean, className?: string): string => classNames(
+  'modal-backdrop',
+  className,
+  { 'modal-backdrop--show': isOpen },
+);
+
+const getModalClasses = (size: Size, isCentered: boolean): string => classNames(
+  'modal',
+  `modal--${size}`,
+  { 'modal--center': isCentered },
+);
+
+type MapChildren = (
+  children: ReactElement<HeaderProps | FooterProps | typeof Content>[]
+) => [string, ReactElement][];
+
+const mapChildren: MapChildren = (children) => children.map((child) => {
+  switch (child.type) {
+    case Header:
+      return ['header', child];
+    case Footer:
+      return ['footer', child];
+    default:
+      return ['content', child];
+  }
+});
+
+export const Modal: FC<ModalProps> & ModalChildren = ({
+  accessibilityDescription,
+  children,
+  className,
+  closeLabel = 'close',
+  id,
+  isCentered = false,
+  isOpen,
+  onClose,
+  size = 'medium',
+}) => {
+  const { header, footer, content } = Object.fromEntries(
+    mapChildren(Array.isArray(children) ? children : [children]),
```

Comment:
```suggestion
    mapChildren(children),
```
If children is an array than the ternary is unnecessary